lib/bech32/test/Codec/Binary/Bech32Spec.hs

```diff
@@ -229,21 +256,13 @@ spec = do
                     corruptedString /= originalString ==>
                         (T.length corruptedString === T.length originalString)
                         .&&.
-                        (result `shouldBe` Left
-                            StringToDecodeMissingSeparatorChar)
-                        .||.
-                        (result `shouldBe` Left
-                            (StringToDecodeContainsInvalidChars []))
-                        .||.
-                        (result `shouldBe` Left
-                            (StringToDecodeContainsInvalidChars
-                                [CharPosition index]))
+                        (result `shouldSatisfy` isLeft)
```

Comment:
If we wanted to be slightly more precise, we could write:
```hs
let result = Bech32.decode corruptedString
let resultMatchesExpectations = \case 
        Left StringToDecodeTooShort                 -> True   
        Left StringToDecodeMissingSeparatorChar     -> True   
        Left (StringToDecodeContainsInvalidChars _) -> True   
        _                                           -> False  
return $ counterexample description $ 
    corruptedString /= originalString ==> 
        (T.length corruptedString === T.length originalString)
        .&&.  
        (result `shouldSatisfy` resultMatchesExpectations) 

```

Reply:
Hi @KtorZ 

> @jonathanknowles I don't think there was an issue here to be honest.... The tests were already limiting their indexes to the worst case (swapping)....

First of all, my apologies if I have misunderstood.

From my understanding, after the merging PR #332, the test suite no longer tests:
* deletion of characters from the ends of strings.
* mutation of characters at the ends of strings.
* insertion of characters at the ends of strings.

This is because the relevant indices will no longer be generated.

Before merging of PR #332, these cases were tested (as far as I can tell). Was there any reason for us to remove coverage of these cases?

 (Apologies again if I have misunderstood!)